### DISCUSSION TITLE
How to work with the User's Locale

### DISCUSSION THREAD

**Discussion by Aziz403** [symfony/symfony#49466] [Q&A]
Hello,
In the doc we have : https://symfony.com/doc/4.3/translation/locale.html
Talks about how to switch locale in **6.0** .
And we can see this is condition to switch locale:
> The custom listener must be called before LocaleListener, which initializes the locale based on the current request. To do so, set your listener priority to a higher value than LocaleListener priority (which you can obtain running the debug:event kernel.request command).

The docs don't explain how to get the User Entity ! 
Because when i tried to get token with _priority higher than 8_, i find it's **null**.
Example:
```
KernelEvents::REQUEST => [
    ['onRequestKernel1',0], // TokenStorage::getToken() returns token, but we need priority higher than 16 :  "set your listener priority to a higher value than LocaleListener priority"
    ['onRequestKernel2',17] // TokenStorage::getToken() returns null !
]
```

![Capture d’écran 2023-02-20 190406](https://user-images.githubusercontent.com/89544871/220326766-450b25af-d29e-410f-8fee-cbcb8febe834.png)

Also same things make not sience for me.
Beacause i tried to create simple example with the same event subscriber and i in this is time can i switch locale with priority lower than 16 ?!!

**Comment by MatTheCat** [comment]
Hello,

> this is time can i switch locale with priority lower than 16 ?!!

You can indeed call `Request::setLocale()` but this won’t have any effect on the services depending on the locale (like the translator). As they got it from the `LocaleAwareListener` you’re supposed to store the correct locale into the request before that.

As you maybe noticed this makes setting the locale from the authenticated user impossible because your subscriber’s priority cannot be lower than `8` and higher than `16`. This is why there is a dedicated section in the documentation about it: https://symfony.com/doc/4.3/session/locale_sticky_session.html#setting-the-locale-based-on-the-user-s-preferences

Note that if you’re using Symfony ≥ 6.1 you can leverage the [`LocaleSwitcher`](https://symfony.com/blog/new-in-symfony-6-1-locale-switcher).
